cirq/study/resolver.py

```diff
@@ -54,8 +55,8 @@ def value_of(
         """
         if isinstance(value, str):
             return self.param_dict.get(value, Symbol(value))
-        if isinstance(value, Symbol):
-            return self.param_dict.get(value.name, value)
+        if isinstance(value, sympy.mul.Mul):
```

Comment:
This is almost certainly incorrect. There could be arbitrarily complicated `sympy` expressions. You should use substitution: https://docs.sympy.org/latest/tutorial/basic_operations.html#substitution

Reply:
Thanks for letting me know about substitutions. I was looking for a way to generalize this and just using `Symbol`  wasn't working.